src/auth/token-verifier.ts

```diff
@@ -185,7 +178,7 @@ export class FirebaseTokenVerifier {
     if (!fullDecodedToken) {
       errorMessage = `Decoding ${this.tokenInfo.jwtName} failed. Make sure you passed the entire string JWT ` +
         `which represents ${this.shortNameArticle} ${this.tokenInfo.shortName}.` + verifyJwtTokenDocsMessage;
-    } else if (typeof header.kid === 'undefined' && this.algorithm !== 'none') {
+    } else if (!useEmulator() && typeof header.kid === 'undefined') {
```

Comment:
May be call this function once in the constructor and assign the result to a private member? No need to keep calling it repeatedly afaict.

Reply:
If we keep this as part of state, it won't correctly respond if `process.env.FIREBASE_AUTH_EMULATOR_HOST` is changed between two calls. This is something we do for unit tests (including the new one added according to your suggestion) and I think developers will probably do this too and get confused. I'd say just keep it since this function just delegates to reading `process.env` and isn't expensive at all.

Reply:
Let's at least make sure that a single invocation of the method uses the same value. If the env variable changes while the method is being executed, different parts of the method will see different values (unlikely scenario, but nevertheless we should implement strong semantics). Call `useEmulator()` once and assign the result to a local `const`, and use it everywhere in the method body. Pass the value to any other private helper methods as needed.

Reply:
Done.